instant-distance-py/src/lib.rs

```diff
@@ -414,73 +413,42 @@ impl Neighbor {
     }
 }
 
-#[repr(align(32))]
 #[derive(Clone, Deserialize, Serialize)]
-struct FloatArray(#[serde(with = "BigArray")] [f32; DIMENSIONS]);
+struct FloatArray<M> {
```

Comment:
Feels to me like the change here from parametrizing the metric is separate from moving from a fixed-size array to a `Vec`, so I'd like those changes to be in separate commits.

Also, once this is changed to hold a `Vec`, should probably also change the name, since it's no longer actually an array? (Maybe `FloatVector`?)

And in particular, I'm curious about the performance impact of the addtional allocations (and pointer indirection/cache) of using a `Vec` here over an array. For that we probably need more of a macro-benchmark, do you have a data set you've tried for this?

Reply:
👍 I'll change the name.

I don't have any real dataset, I've been using random numbers so far. I could potentially use the dataset from `translate.py` example or maybe something form ann-benchmark (although from what I've seen those are usually quite big).